haystack_experimental/components/image_converters/document_to_image.py

```diff
@@ -0,0 +1,204 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import mimetypes
+from pathlib import Path
+from typing import Dict, List, Literal, Optional, Tuple, Union
+
+from haystack import Document, component, logging
+from haystack.dataclasses import ByteStream
+
+from haystack_experimental.components.image_converters.file_to_image import ImageFileToImageContent
+from haystack_experimental.components.image_converters.pdf_to_image import PDFToImageContent
+from haystack_experimental.dataclasses.image_content import IMAGE_MIME_TYPES, ImageContent
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class DocumentToImageContent:
+    """
+    A component to convert documents sourced from PDF and image files to ImageContents.
+
+    This component processes a list of Documents and extracts visual content from supported file formats, converting
+    them into ImageContents that can be used for multimodal AI tasks. It handles both direct image files and PDF
+    documents by extracting specific pages as images.
+
+    The component categorizes input documents into three groups:
+    - Image documents: Successfully converted to ImageContent objects
+    - Non-image documents: Documents that don't contain supported image formats
+    - Documents with missing information: Documents lacking required metadata such as `file_path` or `page_number`.
+
+    Documents will be skipped and categorized as non-image documents in the following scenarios:
+    - The `file_path` is not present in the document metadata
+    - The file path does not point to an existing file when combined with `root_path`
+    - The file format is not among the supported image types
+    - For PDF files, the `page_number` is not present in the metadata
+
+    Example:
+        ```python
+        converter = DocumentToImageContent(
+            root_path="/data/documents",
+            detail="high",
+            size=(800, 600)
+        )
+
+        documents = [
+            Document(content="Optional description of image.jpg", meta={"file_path": "image.jpg"}),
+            Document(content="Text content of page 1 of doc.pdf", meta={"file_path": "doc.pdf", "page_number": 1})
+        ]
+
+        result = converter.run(documents)
+        image_documents = result["image_documents"]
+        image_contents = result["image_contents"]
+        ```
+    """
+
+    def __init__(
+        self,
+        *,
+        root_path: Optional[str] = None,
+        detail: Optional[Literal["auto", "high", "low"]] = None,
+        size: Optional[Tuple[int, int]] = None
+    ):
```

Comment:
does it make sense to introduce sth like `file_path_meta_field` to make the source of the image configurable?

Reply:
yup! I like the idea from your `ImageEmbedder` component. I'll do that here too